src/main/java/net/neoforged/neoforge/common/extensions/IBlockStateExtension.java

```diff
@@ -755,4 +755,11 @@ default boolean isEmpty() {
     default BubbleColumnDirection getBubbleColumnDirection() {
         return self().getBlock().getBubbleColumnDirection(self());
     }
+
+    /**
+     * Return true if this block should cause an adjacent face of the given fluid to not render.
+     */
+    default boolean shouldHideAdjacentFluidFace(FluidState adjacentFluid) {
```

Comment:
I think it would make sense to also pass through the occluding `Direction` of the block this is being called on.